packages/open_aea/connections/p2p_libp2p/libp2p_node/libp2p_node.go

```diff
@@ -92,14 +93,26 @@ func main() {
 
 	// persist agent records to file
 	storagePath := agent.RecordStoragePath()
+
 	// libp2p node
 	var node dhtnode.DHTNode
 
 	// Run as a peer or just as a client
 	if nodePortPublic == 0 {
 		// if no external address is provided, run as a client
+
+		// NOTE: Looks like we need to get the change the ledger API
+		//       to be able to detect the ledger id
+
+		// var identity string
+		// if record.LedgerId == "fetchAI" {
+		// 	identity = dhtclient.IdentityFromFetchAIKey(key)
+		// } else {
+		// 	identity = dhtclient.IdentityFromEthereumKey(key)
+		// }
```

Comment:
Yes that would be desirable ultimately so it can be used across both ecosystems.